### ISSUE TITLE
audit/improve a11y

### ISSUE THREAD

**Issue by cezaraugusto** [brave/brave-ui#137]
we should have a11y in mind and make our components a11y-friendly.

just discovered https://ui.reach.tech/ which is a foundation for design systems. We should consider adopting some of their standards as some components are common to us.

this should be considered before v1.0 or at least before brave-core v1.0.
